components/kyma-environment-broker/internal/cls/deprovisioner.go

```diff
@@ -35,15 +35,14 @@ func NewDeprovisioner(storage DeprovisionerStorage, remover InstanceRemover, log
 }
 
 type DeprovisionRequest struct {
-	GlobalAccountID string
-	SKRInstanceID   string
-	Instance        servicemanager.InstanceKey
+	SKRInstanceID string
+	Instance      servicemanager.InstanceKey
 }
 
 func (d *Deprovisioner) Deprovision(smClient servicemanager.Client, request *DeprovisionRequest) error {
-	instance, _, err := d.storage.FindInstance(request.GlobalAccountID)
+	instance, _, err := d.storage.FindByID(request.Instance.InstanceID)
 	if err != nil {
-		return errors.Wrapf(err, "while trying to lookup an instance for global account: %s", request.GlobalAccountID)
+		return errors.Wrapf(err, "while trying to find the cls instance %s", request.Instance.InstanceID)
```

Comment:
if the CLS instance is not found, do not return any error. Maybe we can log it, but it would be nice to call "Deprovision" method twice and do not see any error.

Reply:
I'll note it down, but I'd rather tackle it in our final PR (since we already noticed some other similar issues in other places)